### ISSUE TITLE
camera Projection seems not working.

### ISSUE THREAD

**Issue by chiragchauhan1073** [google/filament#2954]
Hello, i am trying to set default zoom with camera.setProjection but it does not make any effect on my model.
i have no clue what i am doing wrong here, here is my code.

class MainActivity : AppCompatActivity() {
    companion object {
        // Load the library for the utility layer, which in turn loads gltfio and the Filament core.
        init {
            Utils.init()
        }
    }

    private lateinit var surfaceView: SurfaceView
    private lateinit var choreographer: Choreographer
    private val frameScheduler = FrameCallback()
    private lateinit var modelViewer: ModelViewer

    private lateinit var doubleTapDetector: GestureDetector

    private lateinit var engine: Engine

    // A swap chain is Filament's representation of a surface
    private var swapChain: SwapChain? = null
    private lateinit var renderer: Renderer
    private lateinit var uiHelper: UiHelper
    private lateinit var displayHelper: DisplayHelper

    private lateinit var view: View

    // Should be pretty obvious :)
    private lateinit var camera: Camera

    @SuppressLint("ClickableViewAccessibility")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        surfaceView = SurfaceView(this).apply { setContentView(this) }
        choreographer = Choreographer.getInstance()

        modelViewer = ModelViewer(surfaceView)

        displayHelper = DisplayHelper(this)

        surfaceView.setOnTouchListener { _, event ->
            modelViewer.onTouchEvent(event)
            true
        }

        createRenderables()

        modelViewer.view.ambientOcclusion = View.AmbientOcclusion.SSAO

        val bloom = modelViewer.view.bloomOptions
        bloom.enabled = true
        modelViewer.view.bloomOptions = bloom

        camera = modelViewer.camera

        //Here i am trying to apply default zoom 
        val zoom = 5.0
        val aspect = 100.toDouble() / 50.toDouble()
        camera.setProjection(
            Camera.Projection.ORTHO,
            -aspect * zoom, aspect * zoom, -zoom, zoom, 10.0, 10.0
        )

        setupSurfaceView()
        createIndirectLight()
    }

    private fun createRenderables() {
        val buffer = assets.open("models/test.gltf").use { input ->
            val bytes = ByteArray(input.available())
            input.read(bytes)
            ByteBuffer.wrap(bytes)
        }

        modelViewer.loadModelGltfAsync(buffer) { uri -> readCompressedAsset("models/$uri") }
        modelViewer.transformToUnitCube()

        engine = modelViewer.engine
        renderer = engine.createRenderer()
        view = engine.createView()

    }

    private fun setupSurfaceView() {
        uiHelper = UiHelper(UiHelper.ContextErrorPolicy.DONT_CHECK)
        uiHelper.renderCallback = SurfaceCallback()

        // NOTE: To choose a specific rendering resolution, add the following line:
//         uiHelper.setDesiredSize(600, 800)
        uiHelper.attachTo(surfaceView)
    }

    private fun createIndirectLight() {
        val engine = modelViewer.engine
        val scene = modelViewer.scene
        val ibl = "venetian_crossroads_2k"
        readCompressedAsset("envs/$ibl/${ibl}_ibl.ktx").let {
            scene.indirectLight = KtxLoader.createIndirectLight(engine, it)
            scene.indirectLight!!.intensity = 30_000.0f
        }
        readCompressedAsset("envs/$ibl/${ibl}_skybox.ktx").let {
            scene.skybox = KtxLoader.createSkybox(engine, it)
        }

        scene.skybox = Skybox.Builder().color(01f, 01f, 01f, 01f).build(engine)

    }

    private fun readCompressedAsset(assetName: String): ByteBuffer {
        val input = assets.open(assetName)
        val bytes = ByteArray(input.available())
        input.read(bytes)
        return ByteBuffer.wrap(bytes)
    }

    override fun onResume() {
        super.onResume()
        choreographer.postFrameCallback(frameScheduler)
    }

    override fun onPause() {
        super.onPause()
        choreographer.removeFrameCallback(frameScheduler)
    }

    override fun onDestroy() {
        super.onDestroy()
        choreographer.removeFrameCallback(frameScheduler)
    }

    inner class SurfaceCallback : UiHelper.RendererCallback {
        override fun onNativeWindowChanged(surface: Surface) {
            swapChain?.let { engine.destroySwapChain(it) }
            swapChain = engine.createSwapChain(surface, uiHelper.swapChainFlags)
            displayHelper.attach(renderer, surfaceView.display);
        }

        override fun onDetachedFromSurface() {
            displayHelper.detach();
            swapChain?.let {
                engine.destroySwapChain(it)
                // Required to ensure we don't return before Filament is done executing the
                // destroySwapChain command, otherwise Android might destroy the Surface
                // too early
                engine.flushAndWait()
                swapChain = null
            }
        }

        override fun onResized(width: Int, height: Int) {
            val aspect = width.toDouble() / height.toDouble()
            camera.setProjection(45.0, aspect, 0.1, 50.0, Camera.Fov.VERTICAL)

            view.viewport = Viewport(0, 0, width, height)
        }
    }

    inner class FrameCallback : Choreographer.FrameCallback {
        private val startTime = System.nanoTime()
        override fun doFrame(frameTimeNanos: Long) {
            choreographer.postFrameCallback(this)

            modelViewer.animator?.apply {
                if (animationCount > 0) {
                    val elapsedTimeSeconds = (frameTimeNanos - startTime).toDouble() / 1_000_000_000
                    applyAnimation(0, elapsedTimeSeconds.toFloat())
                }
                updateBoneMatrices()
            }

            modelViewer.render(frameTimeNanos)
        }
    }
}

**Comment by prideout** [comment]
Did you check if the projection is getting overwritten in a resize handler?

Note that ModelViewer lets you pass in your own Manipulator object into its constructor, that way you can set up a custom initial zoom factor.  You also do not need to call `transformToUnitCube` if you would rather not auto-scale the model.

By the way, I think you're setting the renderCallback in UiHelper twice: once in ModelViewer and once in your Activity.

You're also changing the perspective camera into an ortho one, not sure if that's intentional.

**Comment by chiragchauhan1073** [comment]
Okay, I tried to apply Manipulator, but after that, camera moves around the screen, but I just want a steady model that will be at a fixed position. I just want to increase its zoom level.

Here is how it behaves with the manipulator,

![camera](https://user-images.githubusercontent.com/24297327/90336404-5a1e3000-dff9-11ea-89f5-0de24ead6c0a.gif)

But this is what i am looking for
i just wants to increase zoom level, so model looks bigger.
![fixed](https://user-images.githubusercontent.com/24297327/90336550-3b6c6900-dffa-11ea-9b33-4616ff3d95d4.gif)
also please note that model is rotating in  left to right and right to left direction. how to make it rotating 360 degree ?

**Comment by prideout** [comment]
I just now tried modifying the Activity in `sample-gltf-viewer` as follows and it works well (no unexpected panning)

```
manip = Manipulator.Builder()
        .targetPosition(0.0f, 0.0f, -4.0f)
        .orbitHomePosition(0.0f, 0.0f, -2.0f)
        .viewport(surfaceView.width, surfaceView.height)
        .build(Manipulator.Mode.ORBIT)

modelViewer = ModelViewer(surfaceView, Engine.create(), manip)
```

The ORBIT manipulator is similar to the default camera in https://sketchfab.com/.  It allows unbounded rotation around the Y axis (dragging left-and-right) and it bounded rotation around the X axis within [-180,+180] (dragging up-and-down).  If this is not natural for your model, you might want to try transforming your model, e.g.:

```
val tm = engine.transformManager
tm.setTransform(tm.getInstance(asset.root), transform.toFloatArray())
```
